packages/kit/src/runtime/client/router.js

```diff
@@ -165,17 +165,15 @@ export class Router {
 				return;
 			}
 
-			history.pushState({}, '', url.href);
-
 			const noscroll = a.hasAttribute('sveltekit:noscroll');
-			this._navigate(url, noscroll ? scroll_state() : null, false, [], url.hash);
+			this._navigate(url, noscroll ? scroll_state() : null, false, [], url.hash, {}, 'pushState');
 			event.preventDefault();
 		});
 
 		addEventListener('popstate', (event) => {
 			if (event.state && this.enabled) {
 				const url = new URL(location.href);
-				this._navigate(url, event.state['sveltekit:scroll'], false, []);
+				this._navigate(url, event.state['sveltekit:scroll'], false, [], url.hash, null, null);
```

Comment:
should we just allow these lass two parameters to be `undefined` so that you don't have to pass anything here?